File: typescript/src/NullFortyTime.ts
import { BaseFortyTime } from "./BaseFortyTime"

export class NullFortyTime extends BaseFortyTime {
  static parse = (_input: any): NullFortyTime => {
    return new NullFortyTime()
  }

  constructor() {
    super(0)
  }

  get value(): null {
    return null
  }

  toString = (): string => {
    return ""
  }

  plus = (other: BaseFortyTime): BaseFortyTime => {
    return other
  }

  minus = (other: BaseFortyTime): BaseFortyTime => {
    return other
  }
}

File: ruby/lib/null_forty_time.rb
# frozen_string_literal: true

class NullFortyTime < FortyTime
  def initialize
    super(0)
  end

  def value; end

  def +(other)
    other
  end

  def -(other)
    other
  end

  def to_s
    ''
  end
end
